### ISSUE TITLE
cabal install regression

### ISSUE THREAD

**Issue by GeorgeCo** [haskell/cabal#9490]
**Describe the bug**

After cabal install --lib, the installed package is hidden and can't be compiled without using -package

**To Reproduce**

Create a file with the line:

import Data.Tuple.Extra

```
$ cabal build ...
```

**Expected behavior**

**System information**
 - Operating system
 - `cabal`, `ghc` versions

**Additional context**

**Comment by gbaz** [comment]
I don't think this is a bug. The package `cabal-installed` was `tuple` but the missing module is not in that package, but instead in `extra`, which was not cabal-installed.

**Comment by GeorgeCo** [comment]
It seems that this could also be solved by
```
cabal install --lib extra
ghc cb.hs
```
Which of the two solutions is best? 

Also I should have mentioned that the error message needs to be improved as it is intended for ghci users. The command for ghc is similar but not the same.

Putting all this here in case others arrive here when encountering that message.

**Comment by Bodigrim** [comment]
@GeorgeCo you do not need to `cabal install` anything. Put 
```haskell
#!/usr/bin/env cabal
{- cabal:
build-depends: base, extra
-}
import Data.Tuple.Extra
```
and run
```
cabal repl cb.hs
```
